### ISSUE TITLE
Summary of what's needed to finish UDD v1.3

### ISSUE THREAD

**Issue by alanepaull** [jiscdev/analytics-udd#159]
This is my review of stuff that we need to do to finish UDD v1.3 - except for minor revisions of codes or small tweaks that are not structural.  In this list, MAJOR means structural or things that will break it; MINOR means not breaking, or very nearly resolved, so comparatively easy and quick to put in place very soon.

MAJOR:
#145 and FE discussions generally:  Confirm extent of inclusion of FE issues in v1.3.
#77 Design and agree the mechanics of systems identifiers for students and staff (LRW implementation issue).
#74 Design and agree field level extensibility mechanics (LRW implementation issue).
#106 Institution-specific code lists (LRW implementation issue).
#105 Agree reference data structures: period, inst_tier, course_location.
#147 Resolve VLE - UDD module/course ID mapping.

MINOR:
#144 Integrate SEQ properties into IDs.
#146 Agree entity descriptions.
#133 Agree LOCSDY.
#132 Agree progression codes for SoCI.

There is also the review of MODE.  This one would be a major issue (owing to complexity of relationships with other properties in 'engagement'); it could be pushed to 2.0, or included in 1.3.

Alan

**Comment by arc12** [comment]
I have reviewed most (all?) of these issues and added a comment to each about what I think is ready for a v1.3.0 in the "asap" timeframe.

**Comment by robwynj** [comment]
Any “big” or significant blockers for getting 1.3.0 out, should be addressed for 1.3.1 (non-breaking blockers/ outstanding issues) or a 1.4.x (potentially “breaking changes” from blockers)

Hope this makes sense!

--
Rob Wyn Jones BSc(Hons) MBCS
ITIL & ISEB Certified; APMG PRINCE2 & APMG AGILE Certified Practitioner

Senior Data & Analytics Integrator - Jisc UK Learning Analytics
Cluster Lead – Jisc Data & Analytics Service

email/ skype: rob.jones@jisc.ac.uk<mailto:rob.jones@jisc.ac.uk> mobile: 07468 727040  web: www.jisc.ac.uk<http://www.jisc.ac.uk>

**Comment by alanepaull** [comment]
Update at Monday 27 February prior to afternoon meeting:

MAJOR:
**#145 and FE discussions generally:** Confirm extent of inclusion of FE issues in v1.3.
Update: We need to agree the suggestion that we don't map Programme of Learning into the UDD structures for v1.3.0, but that we map FE courses to UDD course, so that conceptually these are handled in the same way as for HE.  When we have agreed this, I can put the extra properties for FE into the correct entities (AWARDING_BODY, PRIOR_ATTAINMENT_ENGLISH, PRIOR_ATTAINMENT_MATHS, INITIAL_ASSESSMENT_ENGLISH, INITIAL_ASSESSMENT_MATHS, PREDICTED_OUTCOME, ATTAINMENT_TARGET.

**#77 Design and agree the mechanics of systems identifiers for students** and staff (LRW implementation issue).
DONE

**#74 Design and agree field level extensibility mechanics** (LRW implementation issue).
DONE

**#106 Institution-specific code lists** (LRW implementation issue).
DONE

**#105 Agree reference data structures: period, inst_tier, course_location.**
DONE

**#147 Resolve VLE - UDD module/course ID mapping.**
DONE

MINOR:
#144 Integrate SEQ properties into IDs.
DONE

#146 Agree entity descriptions.
DONE

#133 Agree LOCSDY.
DONE

#132 Agree progression codes for SoCI.
DONE

I believe the only outstanding issue is the FE one indicated at the top of this comment.  Unless anyone knows different!

**Comment by alanepaull** [comment]
Since I posted this, we've had an explosion of issues, admittedly many fairly small.  As we will need to gain agreement, we need to prioritise the ones that are definitely for v1.3.0 (some of which are indicated), and which are for later.

For v1.3.0 we need these, I believe:

#169 Compound keys as unique identifiers for URLs
#170 v1.3.0 Prior/initial - Change Entity? [mainly for FE]
#171 v1.3.0 Prior/initial value space/definition/type [mainly for FE]
#172 v1.3.0 "Primary Keys"
#173 v1.3.0 ASSESS_SEQ_ID
#174 X_MOD_ACADEMIC_YEAR on SoAI
#175 v1.3.0 Notes on SCM.ACTIVE_MEMBERSHIP
#177 v1.3.0 SCM PREDICTED_OUTCOME & ATTAINMENT_TARGET
#167 Dealing with Course Changes

and can park these for 1.3.1 or later (this is currently all of the parked ones):

#160 Multiple personal tutors
#176 Add attribute to see if student came in via clearing to student course membership
#165 Suggested added notes for MOD_CREDITS
#100 Review treatment of SUBJECT
#66 Review MODE with a view to collapsing most categories
#58 Moving MAX_POINTS from student on assessment instance to assessment instance

**Comment by alanepaull** [comment]
I've reviewed the current issues and put an "Outstanding for v1.3.0" label on any that are.  If you're looking at the UDD github, please focus on these ones!

Alan

**Comment by alanepaull** [comment]
All changes for v1.3.0 now complete.  I will now create the v1.3.0 branch (to be frozen) and v1.3.1 (for non-breaking blockers/ outstanding issues).

**Comment by alanepaull** [comment]
v1.3 is now gone.  We have v1.3.0 as the new version with all the changes that we've agreed in it.  This should now form the frozen v1.3.0 for folks to create data against.

I will leave this alone for 24 hours, in case any issues come up that need looking at before we start the v1.3.1 development branch.

**Comment by alanepaull** [comment]
Only 1 outstanding issue for v1.3.0 now.  This is MOD_FIRST_MARK documentation #189.  I'd like to get this cleared up for v1.3.0 if possible.  However, if it could be left till v1.3.1, please let me know.

**Comment by arc12** [comment]
1.3.0 much preferred
